tfe/data_source_outputs.go

```diff
@@ -159,19 +159,28 @@ type outputData struct {
 
 func (d dataSourceOutputs) readStateOutput(ctx context.Context, tfeClient *tfe.Client, orgName, wsName string) (*stateData, error) {
 	log.Printf("[DEBUG] Reading the Workspace %s in Organization %s", wsName, orgName)
-	opts := &tfe.WorkspaceReadOptions{
-		Include: []tfe.WSIncludeOpt{tfe.WSOutputs},
-	}
-	ws, err := tfeClient.Workspaces.ReadWithOptions(ctx, orgName, wsName, opts)
+	ws, err := tfeClient.Workspaces.Read(ctx, orgName, wsName)
 	if err != nil {
 		return nil, fmt.Errorf("Error reading workspace: %w", err)
 	}
 
+	svOpts := &tfe.StateVersionCurrentOptions{
+		Include: []tfe.StateVersionIncludeOpt{tfe.SVoutputs},
+	}
+	sv, err := tfeClient.StateVersions.ReadCurrentWithOptions(ctx, ws.ID, svOpts)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading state version: %w", err)
+	}
+
+	outputs, err := tfeClient.StateVersions.ListOutputs(ctx, sv.ID, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading state version outputs: %w", err)
+	}
```

Comment:
This will error for configurations that are managed by [someone that does not have permissions](https://www.terraform.io/cloud-docs/users-teams-organizations/permissions#general-workspace-permissions) to read a SV or its outputs. So we should attempt a fallback before throwing an error.

Reply:
Which fallback did you have in mind? If you can't read outputs, this data source will be completely nonfunctional. One possibility is that if you can't read state, but you can read outputs, you could use tfeClient.WorkspaceOuputs.ReadCurrent but this is a newish feature of the API. I'm thinking that if you don't have at least workspace admin permission you're not going to get very far with the provider!

Reply:
I was under the impression that we are OK if there is an error if someone does not have permissions for the outputs. 🤔

Reply:
So thinking about this further, I'm wrong in my suggestion here. We should not have a fallback and talk two endpoints with varying permissions and perhaps we may be better served with a separate datasource that requires permissions to read state versions. There is value, I think, in being able to fetch outputs w/o having permissions to read a state version for a given workspace: See the [yellow box](https://www.terraform.io/language/state/remote-state-data#the-terraform_remote_state-data-source)

Merging these changes as is would strip the ability for me to read outputs from a different workspace (of which I might not be a workspace admin for).

Reply:
Thanks for the comment @sebasslash. I decided to change the approach - let me know what you think!